src/arithmetic.jl

```diff
@@ -23,16 +23,18 @@ for T in (:Taylor1, :TaylorN)
     end
 end
 
-==(a::HomogeneousPolynomial, b::HomogeneousPolynomial) = a.coeffs == b.coeffs
-
+function ==(a::HomogeneousPolynomial, b::HomogeneousPolynomial)
+    a.order == b.order && return a.coeffs == b.coeffs
+    return (a.coeffs == zeros(a.coeffs)) && (b.coeffs == zeros(b.coeffs))
```

Comment:
Use `iszero`?

Reply:
Why is this change necessary?

Reply:
Thanks for the suggestion, it surely makes everything more readable. 

I don't recall exactly where, but at some point I encountered a case where I was comparing two vanishing `HomogeneousPolynomials` of different order, and getting `false`, which is not correct. Vanishing `HomogeneousPolynomials` are kind of special...